lib/src/functions/map.dart

```diff
@@ -73,6 +91,26 @@ final _hasKey = _function("has-key", r"$map, $key", (arguments) {
   return SassBoolean(map.contents.containsKey(key));
 });
 
+SassMap _set(SassMap map, List<Value> keys, Value value) {
+  var mutableMap = Map.of(map.contents);
+  var key = keys.first;
+
+  if (keys.length == 1) {
+    mutableMap[key] = value;
+    return SassMap(mutableMap);
+  }
+
+  SassMap nestedMap;
+  try {
+    nestedMap = mutableMap[key].assertMap();
```

Comment:
Once I land https://github.com/sass/dart-sass/pull/1077, you can use `.tryMap()` here to avoid the unnecessary catch.

Reply:
nice!